### ISSUE TITLE
[BUG] TypeError: unsupported operand type(s) for +: 'set' and 'str'

### ISSUE THREAD

**Issue by pranavvp16** [sktime/sktime#6257]
In file `sktime/datasets/_single_problem_loaders.py` the mentioned error is raised as it is trying to concatenate set to str

```python
if name not in list(tsf_all_datasets):
            raise ValueError(
                {name}
                + " is not a valid dataset name. \
                    List of valid dataset names can be found at \
                    sktime.datasets.tsf_dataset_names.tsf_all_datasets"
            )
```
I'll make a PR fixing the issue by using f string

**Comment by fkiraly** [comment]
yes, there should be an f string.

We should also probably test that the message is raised with `pytest.raises`, to avoid this reoccurring later.

**Comment by fkiraly** [comment]
Fixed via #6258
